ArchiSteamFarm/IPC/Startup.cs

```diff
@@ -133,8 +149,48 @@ internal sealed class Startup {
 			}
 		);
 
-		// Use routing for our API controllers, this should be called once we're done with all the static files mess
 #if !NETFRAMEWORK && !NETSTANDARD
+		// Add support for static files from custom plugins (e.g. HTML, CSS and JS)
+		foreach (string staticFilesDirectory in staticFilesDirectorys) {
+			app.UseStaticFiles(
+				new StaticFileOptions {
+					OnPrepareResponse = static context => {
+						if (context.File is { Exists: true, IsDirectory: false } && !string.IsNullOrEmpty(context.File.Name)) {
+							string extension = Path.GetExtension(context.File.Name);
+
+							CacheControlHeaderValue cacheControl = new();
+
+							switch (extension.ToUpperInvariant()) {
+								case ".CSS" or ".JS":
+									// Add support for SRI-protected static files
+									// SRI requires from us to notify the caller (especially proxy) to avoid modifying the data
+									cacheControl.NoTransform = true;
+
+									goto default;
+								default:
+									// Instruct the caller to always ask us first about every file it requests
+									// Contrary to the name, this doesn't prevent client from caching, but rather informs it that it must verify with us first that his cache is still up-to-date
+									// This is used to handle ASF and user updates to WWW root, we don't want from the client to ever use outdated scripts
+									cacheControl.NoCache = true;
+
+									// All static files are public by definition, we don't have any authorization here
+									cacheControl.Public = true;
+
+									break;
+							}
+
+							ResponseHeaders headers = context.Context.Response.GetTypedHeaders();
+
+							headers.CacheControl = cacheControl;
+						}
+					},
+					FileProvider = new PhysicalFileProvider(staticFilesDirectory),
+					RequestPath = "/" + Directory.GetParent(staticFilesDirectory)?.Name
```

Comment:
Why do we need this? According to the doc it should work oob. I believe the correct thing to do is to let plugin decide what it wants to override and under what location, if it wants to override `/index.html` then let it do that.

Reply:
What I mean is, you can create your `plugin/foo/www/foo` and have your stuff under `/foo`, but if you want to replace ASF-ui for example, nothing stops you from doing `plugin/foo/www/index.html`. Let plugin creators decide.

Reply:
No, it won't cause conflicts if people stick to the recommendation and have `www/MyCustomPath` inside, and if they don't then this is their intention, what if I want to have a plugin which overrides one of the ASF-ui files?

Reply:
It can't override ASF-ui files anyway, because ASF-ui files have higher priority (UseStaticFiles() for ASF-ui is declared prior UseStaticFiles()

Reply:
> And what stops us from changing it...? 🤔

Nothing, but what's the point of doing this when you can just overwrite the original ASF-ui files?

Reply:
It's exactly the same point as making this PR instead of just adding files to www folder.

Reply:
> It's exactly the same point as making this PR instead of just adding files to www folder.

Hmm... Yes, in principle it is fair.
@JustArchi what will be your final decision?

Reply:
> Nothing, but what's the point of doing this when you can just overwrite the original ASF-ui files?

What's the point of this PR if you can just overwrite the original ASF-ui files? 🤔

Reply:
> What's the point of this PR if you can just overwrite the original ASF-ui files? 🤔

_As I said before_ - "Yes, in principle it is fair."

> We should be as flexible as possible if we decide to make life of ASF plugin creators easier. I don't see a single reason why we'd enforce some fixed path on them instead of letting them decide. If somebody wants `/plugin/asdasd/` he can already do that, if somebody else wants `/ryzhe` he can do that too, and if somebody wants it under `/index2.html` likewise. Be flexible, no reason not to.

In this case, in my opinion, it would be correct to create an additional interface for plugins with a web interface, such as `IPluginWithGUI`, with a field for specifying the preferred web path of the plugin, for example:
```csharp
public interface IPluginWithGUI : IPlugin {
    string WebPath { get; }
}
```

**That's why this is a good idea:**

1. If only because now ASF officially supports plugins with a web interface.
2. Providing a request path allows us to use `UseDefaultFiles()` for plugins, which is very important for user experience.
3. Plugin developers can still set the "`/`" web path to override ASF-ui files.
4. This interface allows us to standardize the way plugins communicate that they have a web interface, for example to provide this information to ASF-ui.

Reply:
I don't believe we need to call default files more than once when using `/`, it should still work for all subdirectories, I checked the source code for that. We can just remove that whole request path mess and let plugin creators decide what pathing they want to use, ASF-ui doesn't need to know about it either, especially because we allow plugins to override its files.

Reply:
> I don't believe we need to call default files more than once when using `/`, it should still work for all subdirectories, I checked the source code for that. We can just remove that whole request path mess and let plugin creators decide what pathing they want to use, ASF-ui doesn't need to know about it either, especially because we allow plugins to override its files.

@JustArchi As far as I remember, I tested it at the beginning of the PR and it doesn't work.  I'll test it out tonight when I get a chance.

I think it would be good to provide the plugin url for ASF-ui as an example to display a link to their web interface on the plugins page.

Reply:
Most plugins will most likely not override the ASF-ui interface.  And links to plugins in ASF-ui will have a very positive effect on user experience.

Reply:
That's a question to ASF-ui devs. ASF-ui already can check for existing web interface and add the link to it.

Reply:
> That's a question to ASF-ui devs. ASF-ui already can check for existing web interface and add the link to it.

How? 👀 Ç

Reply:
> I don't believe we need to call default files more than once when using `/`, it should still work for all subdirectories, I checked the source code for that.

@JustArchi We need to call it for every subdirectory, `/` only works for the root path, I just tested that.